drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -672,29 +682,182 @@ static int sys_mm_drv_mm_init(const struct device *dev)
 	 * find virtual address range which are unused
 	 * in the system
 	 */
-	uintptr_t unused_l2_start_aligned =
-		ROUND_UP(POINTER_TO_UINT(&unused_l2_sram_start_marker),
-					 CONFIG_MM_DRV_PAGE_SIZE);
-
-	if (unused_l2_start_aligned < L2_SRAM_BASE ||
-	    unused_l2_start_aligned > L2_SRAM_BASE + L2_SRAM_SIZE) {
+	if (L2_SRAM_BASE + L2_SRAM_SIZE < UNUSED_L2_START_ALIGNED ||
+	    L2_SRAM_BASE > UNUSED_L2_START_ALIGNED) {
 		__ASSERT(false,
 			 "unused l2 pointer is outside of l2 sram range %p\n",
-			 unused_l2_start_aligned);
+			 UNUSED_L2_START_ALIGNED);
 		return -EFAULT;
 	}
 
 	/*
 	 * Unmap all unused physical pages from the entire
 	 * virtual address space to save power
 	 */
-	size_t unused_size = CONFIG_KERNEL_VM_BASE + CONFIG_KERNEL_VM_SIZE -
-			     unused_l2_start_aligned;
+	size_t unused_size = L2_SRAM_BASE + L2_SRAM_SIZE - UNUSED_L2_START_ALIGNED;
```

Comment:
Here you reverted change for virtual memory Fix. Virtual memory space is much higher than physical one.